### PR TITLE
docs: add ADR on P2P refactor scope

### PR DESCRIPTION
**Pull request by erikgrinaker**
[Rendered](https://github.com/tendermint/tendermint/blob/erik/adr-p2p-refactor/docs/architecture/adr-061-p2p-refactor-scope.md)

This summarizes recent discussions on the scope of the upcoming P2P refactor.

### CODE REVIEW

**Review by tessr — APPROVED** [review]
This is a great synthesis!

File: docs/architecture/adr-061-p2p-refactor-scope.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    * Peer prioritization. [#2860](https://github.com/tendermint/tendermint/issues/2860) [#2041](https://github.com/tendermint/tendermint/issues/2041)
+* Pluggable transports, with `MConnection` as one implementation. [#5587](https://github.com/tendermint/tendermint/issues/5587) [#2430](https://github.com/tendermint/tendermint/issues/2430) [#805](https://github.com/tendermint/tendermint/issues/805)
+* Improved peer address handling.
+    * Address book refactor. [#4848](https://github.com/tendermint/tendermint/issues/4848) [#2661](https://github.com/tendermint/tendermint/issues/2661)
+    * Transport-agnostic peer addressing. [#5587](https://github.com/tendermint/tendermint/issues/5587) [#3782](https://github.com/tendermint/tendermint/issues/3782) [#3692](https://github.com/tendermint/tendermint/issues/3692)
```
Comment (marbar3778) [inline docs/architecture/adr-061-p2p-refactor-scope.md:48]: Would this be a protocol breaking change?

I assume we would need to change the address to somehow notify other nodes if its tcp or udp?

**Review by marbar3778 — APPROVED** [review]

File: docs/architecture/adr-061-p2p-refactor-scope.md (hunk 1 above)
Comment (erikgrinaker) [reply docs/architecture/adr-061-p2p-refactor-scope.md:48]: It depends on a lot of details, but I think we should be able to come up with internal Go abstractions for this first while only supporting TCP/IP address/port numbers in the protocol, and then modify the protocol to allow other address types later.

**Comment by jackzampolin** [comment]
This is the thing I'm most excite about in the whole of the cosmos

**Review by alexanderbez — APPROVED** [review]
Looks amazing!

**Pull request merged**